drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1580 @@
+local function max_year_schedule_of_user_handler(driver, device, ib, response)
+  device:emit_event(capabilities.lockSchedules.yearDaySchedulesPerUser(ib.data.value, {visibility = {displayed = false}}))
+end
+
+-- Capability Handler
+-----------------
+-- Lock/Unlock --
+-----------------
+local function handle_lock(driver, device, command)
+  local ep = device:component_to_endpoint(command.component)
+  local cota_cred = device:get_field(lock_utils.COTA_CRED)
+  if cota_cred then
+    device:send(
+      DoorLock.server.commands.LockDoor(device, ep, cota_cred)
+    )
+  else
+    device:send(DoorLock.server.commands.LockDoor(device, ep))
+  end
+end
+
+local function handle_unlock(driver, device, command)
+  local ep = device:component_to_endpoint(command.component)
+  local cota_cred = device:get_field(lock_utils.COTA_CRED)
+  if cota_cred then
+    device:send(
+      DoorLock.server.commands.UnlockDoor(device, ep, cota_cred)
+    )
+  else
+    device:send(DoorLock.server.commands.UnlockDoor(device, ep))
+  end
+end
+
+----------------
+-- User Table --
+----------------
+local function add_user_to_table(device, userIdx, usrType)
+  -- Get latest user table
+  local user_table = device:get_latest_state(
+    "main",
+    capabilities.lockUsers.ID,
+    capabilities.lockUsers.users.NAME
+  ) or {}
+  local new_user_table = {}
+
+  -- Recreate user table
+  for index, entry in pairs(user_table) do
+    table.insert(new_user_table, entry)
+  end
+
+  -- Add new entry to table
+  table.insert(new_user_table, {userIndex = userIdx, userType = usrType})
+  device:emit_event(capabilities.lockUsers.users(new_user_table, {visibility = {displayed = false}}))
+end
+
+local function update_user_in_table(device, userIdx, usrType)
+  -- Get latest user table
+  local user_table = device:get_latest_state(
+    "main",
+    capabilities.lockUsers.ID,
+    capabilities.lockUsers.users.NAME
+  ) or {}
+  local new_user_table = {}
+
+  -- Recreate user table
+  local i = 0
+  for index, entry in pairs(user_table) do
+    if entry.userIndex == userIdx then
+      i = index
+    end
+    table.insert(new_user_table, entry)
+  end
+
+  -- Update user entry
+  if i ~= 0 then
+    new_user_table[i].userType = usrType
+    device:emit_event(capabilities.lockUsers.users(new_user_table, {visibility = {displayed = false}}))
+  end
+end
+
+local function delete_user_from_table(device, userIdx)
+  -- If User Index is ALL_INDEX, remove all entry from the table
+  if userIdx == ALL_INDEX then
+    device:emit_event(capabilities.lockUsers.users({}, {visibility = {displayed = false}}))
+    return
+  end
+
+  -- Get latest user table
+  local user_table = device:get_latest_state(
+    "main",
+    capabilities.lockUsers.ID,
+    capabilities.lockUsers.users.NAME
+  ) or {}
+  local new_user_table = {}
+
+  -- Recreate user table
+  for index, entry in pairs(user_table) do
+    if entry.userIndex ~= userIdx then
+      table.insert(new_user_table, entry)
+    end
+  end
```

Comment:
Similar to above, would it not be more efficient to call this:
```suggestion
  -- Recreate user table
  local i = 0
  for index, entry in pairs(user_table) do
    if entry.userIndex ~= userIdx then
      i = index
      break
    end
  end

new_user_table = table.remove(user_table, i)
```
or something like that?

Reply:
I modified it. Thank you~!